src/numeric/processes.jl

```diff
@@ -8,36 +10,62 @@ abstract DiscreteProcess <: AbstractProcess
 abstract ContinuousProcess <: AbstractProcess
 abstract IIDExogenous <: AbstractProcess
 
-abstract AbstractDiscretizedProcess <: DiscreteProcess
+abstract AbstractDiscretizedProcess # <: DiscreteProcess
 
+###
+### Discretized process
+###
 
-# discretized process
-
+# date-t grid has a known structure
 type DiscretizedProcess <: AbstractDiscretizedProcess
-    nodes::Matrix{Float64}
-    integration_nodes::Array{Matrix{Float64}}
-    integration_weights::Array{Vector{Float64}}
+    grid::Grid
+    integration_nodes::Array{Matrix{Float64},1}
+    integration_weights::Array{Vector{Float64},1}
 end
 
-n_nodes(dp::DiscretizedProcess) = size(dp.nodes, 1)
+n_nodes(dp::DiscretizedProcess) = n_nodes(dp.grid)
+node(dp::DiscretizedProcess, i) = node(dp.grid,i)
 n_inodes(dp::DiscretizedProcess, i::Int) = size(dp.integration_nodes[i], 1)
-inodes(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_nodes[i][j, :]
-iweights(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_weights[i][j]
-node(dp::DiscretizedProcess, i) = dp.nodes[i, :]
+inode(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_nodes[i][j, :]
+iweight(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_weights[i][j]
+
+# date-t grid is unstructured
+type DiscreteMarkovProcess <: AbstractDiscretizedProcess
+    grid::UnstructuredGrid
+    transitions::Matrix{Float64}
+    values::Matrix{Float64}
+end
+
+DiscreteMarkovProcess(transitions::Matrix{Float64}, values::Matrix{Float64}) =
+    DiscreteMarkovProcess(UnstructuredGrid(transitions), transitions, values)
+
+n_nodes(dp::DiscreteMarkovProcess) = size(dp.values, 1)
+n_inodes(dp::DiscreteMarkovProcess, i::Int) = size(dp.values, 1)
+inode(dp::DiscreteMarkovProcess, i::Int, j::Int) = dp.values[j, :]
+iweight(dp::DiscreteMarkovProcess, i::Int, j::Int) = dp.transitions[i,j]
+node(dp::DiscreteMarkovProcess, i) = dp.values[i, :]
+
 
+# date-t grid is empty
 
 type DiscretizedIIDProcess <: AbstractDiscretizedProcess
     # nodes::Matrix{Float64}
+    grid::EmptyGrid
     integration_nodes::Matrix{Float64}
     integration_weights::Vector{Float64}
 end
 
+DiscretizedIIDProcess(x,w) = DiscretizedIIDProcess(EmptyGrid(), x, w)
+
 n_nodes(dp::DiscretizedIIDProcess) = 0
 n_inodes(dp::DiscretizedIIDProcess, i::Int) = size(dp.integration_nodes, 1)
-inodes(dp::DiscretizedIIDProcess, i::Int, j::Int) = dp.integration_nodes[j, :]
-iweights(dp::DiscretizedIIDProcess, i::Int, j::Int) = dp.integration_weights[j]
+inode(dp::DiscretizedIIDProcess, i::Int, j::Int) = dp.integration_nodes[j, :]
+iweight(dp::DiscretizedIIDProcess, i::Int, j::Int) = dp.integration_weights[j]
 node(dip::DiscretizedIIDProcess, i) = zeros(n_inodes(dip, 1))
 
+
+
+
 # Normal law
 
 immutable MvNormal <: IIDExogenous
```

Comment:
What do you mean by wrap ?
Subtyping Distributions.MvNormal ? Or synchronize the fields of Dolo.MvNormal ? Some kind of multiple inheritance (which apparently won't make it in to 1.0) ?
In principle I'm not against using the Distributions object rather directly, rather the opposite, but I was attempting to proceed cautiously. There are small details I'm not completely sure about. Like do we officially require users to use unicode symbols in yaml file ? In some environments it is quite a pain...